File: Skills.Shared/V1/Response/CharacterResponse.cs
using Newtonsoft.Json;
﻿using System.Text.Json.Serialization;

namespace Skills.Shared.V1.Response;

public class CharacterResponse : BaseResponseEntity
{
    [JsonPropertyName("priority")]
    public int Priority { get; set; }

    [JsonPropertyName("buildName")]
    public string BuildName { get; set; }
    [JsonPropertyName("startingDate")]
    public DateTime StartingDate { get; set; }
    [JsonPropertyName("skills")]
    public IEnumerable<SkillResponse>? Skills { get; set; }
    [JsonPropertyName("story")]
    public string? Story { get; set; }
    [JsonPropertyName("photoId")]
    public Guid? PhotoId { get; set; }
    [JsonPropertyName("photo")]
    public FileEntityResponse? Photo { get; set; }
}

File: Skills/ClientApp/src/models/ResponseModels/CharacterResponse.ts
import FileModel from "../FileModel"
import CharacterSkillResponse from "./CharacterSkillResponse"
import SkillSetResponse from "./SkillSetResponse"

export default interface CharacterResponse {
    id?: string
    priority: number
    buildName: string
    startingDate: Date
    skills: CharacterSkillResponse[]
    story?: string
    photoId?: string
    photo?: FileModel
    skillSet: SkillSetResponse
}